### ISSUE TITLE
PR validation workflows fail on flake8/lint B909 editing a loop's mutable iterable often leads to unexpected results/bugs

### ISSUE THREAD

**Issue by sebastien-lenard** [landlab/landlab#1913]
Hi!

PR validation workflow fails on lint and prevents passing the PRs. The fail is on tens of files not respecting the flake rule B909 editing a loop's mutable iterable often leads to unexpected results/bugs (see below)

Is it possible to temporarily deactivate that rule B909? Because updating all the files concerned by the B909 will probably takes some time...

@mcflugen 

```
landlab/ca/boundaries/hex_lattice_tectonicizer.py:1022:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/ca/boundaries/hex_lattice_tectonicizer.py:1028:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/ca/boundaries/hex_lattice_tectonicizer.py:1029:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/ca/celllab_cts.py:577:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/cmd/authors.py:200:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/cmd/authors.py:200:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/cmd/authors.py:202:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/cmd/authors.py:202:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/cmd/authors.py:268:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/cmd/authors.py:270:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/chi_index/channel_chi.py:444:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/depression_finder/lake_mapper.py:1066:21: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/depression_finder/lake_mapper.py:1068:25: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/depression_finder/lake_mapper.py:1072:25: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/depression_finder/lake_mapper.py:1090:25: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/depression_finder/lake_mapper.py:1092:29: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/depression_finder/lake_mapper.py:1096:29: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/depression_finder/lake_mapper.py:[124](https://github.com/landlab/landlab/actions/runs/8775769184/job/24078484165?pr=1911#step:4:125)1:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/flow_accum/flow_accum_to_n.py:161:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/flow_accum/flow_accum_to_n.py:187:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/flow_accum/flow_accumulator.py:899:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/flow_accum/flow_accumulator.py:986:21: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/flow_director/flow_direction_dinf.py:367:9: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/flow_director/flow_direction_dinf.py:368:9: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/flow_director/flow_direction_dinf.py:369:9: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/flow_director/flow_direction_dinf.py:370:9: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/flow_director/flow_direction_dinf.py:371:9: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/flow_director/flow_direction_dinf.py:372:9: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/gravel_river_transporter/gravel_river_transporter.py:20:9: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/gravel_river_transporter/gravel_river_transporter.py:24:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/gravel_river_transporter/gravel_river_transporter.py:25:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/gravel_river_transporter/gravel_river_transporter.py:26:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/gravel_river_transporter/gravel_river_transporter.py:513:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/hack_calculator/hack_calculator.py:325:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/hand_calculator/hand_calculator.py:196:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/lake_fill/lake_fill_barnes.py:115:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/lake_fill/lake_fill_barnes.py:737:21: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/lake_fill/lake_fill_barnes.py:[127](https://github.com/landlab/landlab/actions/runs/8775769184/job/24078484165?pr=1911#step:4:128)7:21: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/lake_fill/lake_fill_barnes.py:1300:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/lake_fill/lake_fill_barnes.py:1301:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/lake_fill/lake_fill_barnes.py:1302:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/landslides/landslide_probability.py:622:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/landslides/landslide_probability.py:623:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/landslides/landslide_probability.py:624:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/lateral_erosion/lateral_erosion.py:460:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/lateral_erosion/lateral_erosion.py:474:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/lateral_erosion/lateral_erosion.py:522:21: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/lateral_erosion/lateral_erosion.py:525:21: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/lateral_erosion/lateral_erosion.py:591:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/lateral_erosion/lateral_erosion.py:602:21: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/lateral_erosion/lateral_erosion.py:682:25: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/lateral_erosion/lateral_erosion.py:685:25: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/lithology/lithology.py:641:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/network_sediment_transporter/bed_parcel_initializers.py:670:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/normal_fault/normal_fault.py:274:21: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/normal_fault/normal_fault.py:356:21: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/normal_fault/normal_fault.py:369:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/normal_fault/normal_fault.py:409:21: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/normal_fault/normal_fault.py:412:21: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/normal_fault/normal_fault.py:430:21: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/normal_fault/normal_fault.py:433:21: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/profiler/channel_profiler.py:751:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/profiler/channel_profiler.py:765:21: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/profiler/channel_profiler.py:769:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/profiler/trickle_down_profiler.py:342:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/sink_fill/fill_sinks.py:215:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/space/space.py:607:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/species_evolution/species_evolver.py:381:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/transport_length_diffusion/transport_length_hillslope_diffusion.py:287:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/components/transport_length_diffusion/transport_length_hillslope_diffusion.py:301:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/core/utils.py:690:25: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/graph/sort/sort.py:762:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/grid/base.py:586:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/grid/create_network.py:682:9: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/grid/raster_set_status.py:78:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/io/netcdf/dump.py:111:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/io/netcdf/dump.py:139:9: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/io/netcdf/load.py:78:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/io/netcdf/read.py:198:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/io/netcdf/write.py:430:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/io/netcdf/write.py:474:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/io/netcdf/write.py:708:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/io/shapefile/read_shapefile.py:383:9: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/io/shapefile/read_shapefile.py:454:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/layers/eventlayers.py:1011:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/utils/fault_facet_finder.py:151:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/utils/fault_facet_finder.py:152:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/utils/fault_facet_finder.py:153:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/utils/flow__distance.py:219:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/utils/watershed.py:90:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/utils/watershed.py:96:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/utils/watershed.py:216:9: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
landlab/utils/window_statistic.py:229:9: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
noxfile.py:191:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
scripts/introspect_components.py:144:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
scripts/introspect_components.py:144:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
scripts/introspect_components.py:149:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
tests/components/lateral_erosion/test_latero.py:37:9: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
tests/components/lateral_erosion/test_latero.py:[130](https://github.com/landlab/landlab/actions/runs/8775769184/job/24078484165?pr=1911#step:4:131):9: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
tests/components/lateral_erosion/test_latero.py:197:9: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
tests/components/lateral_erosion/test_latero.py:262:9: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
tests/components/test_components.py:160:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
tests/components/test_components.py:160:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
tests/components/test_components.py:180:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
tests/graph/structured_quad/test_quad.py:200:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
```

**Comment by mcflugen** [comment]
@sebastien-lenard You can ignore the error for now but I think it needs fixing. I can start that in a separate pull request.

**Comment by sebastien-lenard** [comment]
Hi @mcflugen and thank you.

 I could take in charge the dev to fix. I would just need help/advice on how flake8 expects to correct the B909 error.. I found very little on internet...

**Comment by mcflugen** [comment]
@sebastien-lenard I'm looking at this now and it looks to me that most (maybe even all) of the found lint lines are false positives.

**Comment by mcflugen** [comment]
Yup, I think that is the case. See https://github.com/PyCQA/flake8-bugbear/issues/467.

In the meantime, you can exclude the (likely) buggy version of *flake8-bugbear* in your ``.pre-commit-config.yaml`` file:
```yaml
additional_dependencies:
    - flake8-bugbear!=24.4.21
```

**Comment by mcflugen** [comment]
Closed with #1917.
